### ISSUE TITLE
'last quality' vs 'last bitrate'

### ISSUE THREAD

**Issue by harshita1804** [hongzimao/pensieve#112]
What is the difference between the two?... is'nt the last quality proportional to the last bitrate?

**Comment by hongzimao** [comment]
They should mean the same thing - i.e., the bitrate from the previous action. We were just using different names at different points during the project development. Sorry for the confusion.

**Comment by harshita1804** [comment]
My question was for MPC server, I see there to compute smoothness_diffs, both these values are used which would be equal right?

**Comment by hongzimao** [comment]
They should be equal. You can also try logging the values of these variables just to make sure.

**Comment by harshita1804** [comment]
yeah, they are equal, but what confuses me is the optimization in the MPC paper, takes the current bitrate and the next bitrate to compute the smoothness diff, so won't using equal value simply make smoothness diff = 0 for all entries?

**Comment by harshita1804** [comment]
I also had an implementation question, I was trying to implement robust mpc, but my mpc always chooses the highest or the lowest bitrate, any ideas on how to solve this issue, I have been stuck with it for the last 2 days. Any suggestion will be highly appreciated

**Comment by hongzimao** [comment]
We also provided a robust MPC implementation in https://github.com/hongzimao/pensieve/blob/master/rl_server/robust_mpc_server.py#L200 Perhaps you can check the output from your implementation and compare with ours?

**Comment by harshita1804** [comment]
I did actually, so the problem is my algo keeps choosing the highest bitrate, I feel there is something wrong with my reward function

**Comment by hongzimao** [comment]
It's likely - I think you have a few viable ideas to try out. Why don't you log the reward from our implementation and compare with yours? robustMPC should be deterministic and all results should be reproducible. Debugging this down might be tedious but doing it right can be interesting.

**Comment by harshita1804** [comment]
yeah, it worked actually, I was using the linear reward, but disregarded the fact that my bitrates were in bits/sec which were huge so the reward was very high converting to mbps helped.. thankyou for your time!
